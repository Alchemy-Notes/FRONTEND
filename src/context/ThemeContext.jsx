import { createContext, useContext, useState } from 'react';

const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState(false);

  return (
    <ThemeContext.Provider value={{ theme, setTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};

const useTheme = () => {
  const context = useContext(ThemeContext);

  if (context === undefined) {
    throw new Error('useTheme must be used within a ThemeContext Provider');
  }

  return context;
};

export { ThemeProvider, useTheme };
