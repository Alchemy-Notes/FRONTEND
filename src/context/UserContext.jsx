import { createContext, useContext, useEffect, useState } from 'react';
import { getMe } from '../services/auth';
import { getUserNotes } from '../services/notes';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState(null);
  const [notes, setNotes] = useState(null);
  const [tree, setTree] = useState(null);

  useEffect(() => {
    async function get() {
      const result = await getMe();
      setUser(result);
    }
    get();
  }, []);

  useEffect(() => {
    async function getNotes() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) getNotes();
  }, [user]);

  return user && notes ? (
    <UserContext.Provider
      value={{ user, setUser, notes, setNotes, tree, setTree }}
    >
      {children}
    </UserContext.Provider>
  ) : (
    <></>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserContext Provider');
  }

  return context;
};

export { UserProvider, useUser };
