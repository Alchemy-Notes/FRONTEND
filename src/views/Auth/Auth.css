.logo {
  /* grid-column-start: 1;
  grid-column-end: 4;
  grid-row-start: 1;
  grid-row-end: 2; */
  grid-area: logo;
  /* width: 70vw; */
  max-height: 20vh;
  margin-left: 15%;
}
.welcome {
  /* grid-column-start: 1;
  grid-column-end: 2;
  grid-row-start: 2;
  grid-row-end: 3; */
  grid-area: welcome;
}

.githubButton {
  /* grid-column-start: 1;
  grid-column-end: 2;
  grid-row-start: 3;
  grid-row-end: 4; */
  grid-area: githubButton;
}

.userForm {
  /* grid-column-start: 1;
  grid-column-end: 2;
  grid-row-start: 4;
  grid-row-end: 5; */
  grid-area: userForm;
}

.swapLogin {
  /* grid-column-start: 1;
  grid-column-end: 2;
  grid-row-start: 5;
  grid-row-end: 6; */
  grid-area: swapLogin;
}

.computer {
  grid-area: computer;
  height: 45vh;
}

.container {
  display: grid;
  text-align: center;
  align-items: center;
  grid-template-columns: auto;
  grid-template-rows: auto;
  grid-template-areas:
    'logo logo logo'
    'welcome . .'
    'githubButton . .'
    'userForm . .'
    'swapLogin computer computer'
    '. computer computer';
}

body {
  margin: 0;
  background-color: #e9bdb3;
  background-image: url('../../../public/assets/NotedBG.png');
  background-position: bottom right;
  background-repeat: no-repeat;
  background-size: cover;
}
