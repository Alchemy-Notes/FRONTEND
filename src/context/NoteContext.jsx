import { createContext, useContext, useEffect, useState } from 'react';
import { getUserNotes } from '../services/notes';
import { useUser } from './UserContext';

const NoteContext = createContext();

const NoteProvider = ({ children }) => {
  const [notes, setNotes] = useState([]);
  const { user } = useUser();

  useEffect(() => {
    async function get() {
      const result = await getUserNotes({
        userId: user.id,
        query: { type: 'favorite' },
      });
      setNotes(result);
    }
    if (user && user.id) get();
  }, []);

  return (
    <NoteContext.Provider value={{ notes, setNotes }}>
      {children}
    </NoteContext.Provider>
  );
};

const useNotes = () => {
  const context = useContext(NoteContext);

  if (context === undefined) {
    throw new Error('useNotes must be used within a NoteContext Provider');
  }

  return context;
};

export { NoteProvider, useNotes };
