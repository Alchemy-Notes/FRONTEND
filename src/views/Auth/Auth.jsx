import { Link, useHistory } from 'react-router-dom';
import GithubLogin from '../../components/GithubLogin/GithubLogin';
import UserForm from '../../components/UserForm/UserForm';
import { useUser } from '../../context/UserContext';
import { codeExchange, signinUser, signupUser } from '../../services/auth';
import { useEffect, useState } from 'react';

export default function Auth({ isSigningUp = false }) {
  const history = useHistory();
  const { setUser } = useUser();
  const [error, setError] = useState(null);

  useEffect(() => {
    const url = new URL(window.location.href);
    const params = url.searchParams;
    const code = params.get('code');
    if (code) {
      try {
        codeExchange(code).then((res) => setUser(res.name));
        history.push('/notes');
      } catch (error) {
        setError(error.message);
      }
    }
  }, []);

  // pull staging?

  const handleSubmit = async (formState) => {
    try {
      if (isSigningUp) {
        const user = await signupUser(formState);
        setUser(user);
        history.push('/notes');
      } else {
        const user = await signinUser(formState);
        setUser(user);
        history.push('/notes');
      }
    } catch (error) {
      setError(error.message);
    }
  }; //TODO: write handleSubmit function

  return (
    <section
    // className={styles.authForm}
    >
      <h2>{isSigningUp ? 'Welcome' : 'Welcome back!'}</h2>
      <br />

      <GithubLogin
        label={isSigningUp ? 'Sign up with GitHub' : 'Sign in with GitHub'}
        setUser={setUser}
      />
      {error ? <p>{error}</p> : <></>}

      <UserForm
        isSigningUp={isSigningUp}
        onSubmit={handleSubmit}
        label={isSigningUp ? 'Sign Up' : 'Log In'}
      />
      {isSigningUp ? (
        <p>
          Already a user? <Link to="/login">Log In</Link>
        </p>
      ) : (
        <p>
          Not a user? <Link to="/register">Sign Up</Link>
        </p>
      )}
    </section>
  );
}
