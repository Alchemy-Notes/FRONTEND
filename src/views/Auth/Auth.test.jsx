import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom/';
import { ThemeProvider } from '../../context/ThemeContext';
import { UserProvider } from '../../context/UserContext';
import Auth from './Auth';

it('should render our button to log into github', () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <ThemeProvider>
          <Auth />
        </ThemeProvider>
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});
