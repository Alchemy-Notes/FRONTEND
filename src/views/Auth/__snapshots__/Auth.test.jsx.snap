// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our button to log into github 1`] = `
<div>
  <section
    class="container"
  >
    <h2
      class="welcome"
    >
      Welcome back!
    </h2>
    <br />
    <div
      class="githubButton"
    >
      <button
        class="lightButton"
      >
        Sign in with GitHub
      </button>
    </div>
    <form
      class="userForm"
    >
      <fieldset
        class="form"
      >
        <section
          class="formSection"
        >
          <label
            for="username"
          >
            Username 
          </label>
          <input
            class="input"
            id="username"
            name="username"
            type="username"
            value=""
          />
        </section>
        <section
          class="formSection"
        >
          <label
            for="password"
          >
            Password 
          </label>
          <input
            class="input"
            id="password"
            name="password"
            type="password"
            value=""
          />
        </section>
        <button
          class="lightButton"
        >
          Log In
        </button>
        
      </fieldset>
    </form>
    <p
      class="swapLogin"
    >
      Not a user? 
      <a
        href="/register"
      >
        Sign Up
      </a>
    </p>
  </section>
</div>
`;
